client/components/domain-diagnostic/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+export default React.createClass( {
+
+	displayName: 'DomainDiagnostic',
+
+	propTypes: {
+		domain: React.PropTypes.string.isRequired,
+	},
+	getInitialState: function() {
+		return {
+			domainDidFail: false
+		};
+	},
+
+	componentDidMount() {
+		const self = this;
+		const imageChecker = new Image();
+		imageChecker.onerror = function() {
+			self.setState( { domainDidFail: true } );
+		}
+		imageChecker.src = 'http://' + this.props.domain + '/w32.gif';
```

Comment:
Soon we will serve everything over https, I'm not sure that this validation should be done client side anyway.

If we do this server side we can do more comprehensive checks and also consider other things, like scheduling a future notification if it still isn't resolved in n days.